kasa/device_factory.py

```diff
@@ -128,34 +128,6 @@ def _perf_log(has_params: bool, perf_type: str) -> None:
         )
 
 
-def _get_device_type_from_sys_info(info: dict[str, Any]) -> DeviceType:
-    """Find SmartDevice subclass for device described by passed data."""
```

Comment:
This is moved unchanged into the `IotDevice` class **except**: rather than raising an exception at the end it defaults to plug. This is consistent with the `smart` and `smartcamera` logic.